app/helpers.go

```diff
@@ -126,6 +128,12 @@ func httpAuthorize(app *App, userID string, topics []string) (bool, []string, er
 		}
 
 		response, err := client.Do(request)
+		// discard response body
+		if response != nil && response.Body != nil {
+			_, _ = io.Copy(ioutil.Discard, response.Body)
+			_ = response.Body.Close()
+		}
+
 		if err != nil {
```

Comment:
I would put the error check before, so that you wouldn't need to check if response or the body are nil

Reply:
I agree, but the problem is when the error is related to Redirection; `err` will be not nil and `resp.body` will also be non-nil, so it would still leak memory

http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#close_http_resp_body